projects/components/src/navigation/navigation-list.component.ts

```diff
@@ -76,28 +76,26 @@ export class NavigationListComponent implements OnChanges {
   @Output()
   public readonly navItemSelected: EventEmitter<NavItemLinkConfig> = new EventEmitter();
 
+  @Output()
+  public readonly activeItemChange: EventEmitter<NavItemLinkConfig> = new EventEmitter();
+
   public activeItem$?: Observable<NavItemLinkConfig | undefined>;
 
   public constructor(
     private readonly navigationService: NavigationService,
     private readonly activatedRoute: ActivatedRoute,
-    private readonly navListComponentService: NavigationListComponentService,
-    private readonly timeRangeService: TimeRangeService
+    private readonly navListComponentService: NavigationListComponentService
   ) {}
 
   public ngOnChanges(changes: TypedSimpleChanges<this>): void {
     if (changes.navItems) {
       this.navItems = this.navListComponentService.resolveFeaturesAndUpdateVisibilityForNavItems(this.navItems);
 
-      // Initialize the time range service
-      // Depending on FF status, the TR will be either global or page level for the init
       this.activeItem$ = this.navigationService.navigation$.pipe(
         startWith(this.navigationService.getCurrentActivatedRoute()),
         map(() => {
           const activeItem = this.findActiveItem(this.navItems);
-          if (!this.timeRangeService.isInitialized() && activeItem?.timeRangeResolver) {
-            this.timeRangeService.setDefaultTimeRange(activeItem.timeRangeResolver());
-          }
+          this.activeItemChange.emit(activeItem);
```

Comment:
I'm ok with it but would be remiss not to point out - we're using this observable for side effects. If we happened to not read activeItem$ in this component because we didn't need to use it for our own template, it would create a bug where we also stopped emitting.

The alternative however is also not great, since it means we need to subscribe in the component and handle unsubscribing on destroy.

Reply:
Yeah fair concern, i've made a code comment about it for anyone that sees this in the future.